### ISSUE TITLE
Cannot chain onValue and onError to produce "Promise-like" handling of values

### ISSUE THREAD

**Issue by sgerace** [baconjs/bacon.js#382]
So, coming from the Promise world (in particular, [bluebird](https://github.com/petkaantonov/bluebird)), I'm used to writing code like this:

``` javascript
fcnReturningAsyncPromise().then(function (value) {
  // Do something on success
}).catch(function (value) {
  // Do something on failure
});
```

Unfortunately, this does seem to work when `someAsyncFunction` returns Bacon `observables`:

``` javascript
fcnReturningBaconObservable().onValue(function (value) {
  // Do something on success
}).onError(function (value) {
  // Do something on failure
});
```

Instead, I have to do this:

``` javascript
var observable = fcnReturningBaconObservable();
observable.onValue(function (value) {
  // Do something on success
})
observable.onError(function (value) {
  // Do something on failure
});
```

So I'm guessing it is just an issue of not returning `this` from something (or returning the wrong `this` perhaps). I know it isn't a big deal, and this really doesn't crop up in too many places in the code (since most code just interacts and manipulates the `observables` directly); however, I was curious whether or not this was as intended (for reasons unbeknownst to me). 

As a little background (and to justify my sudden interest in Bacon.js), I'm working on a new project that requires use of a more reactive programming style and after playing around with both RxJS and Bacon.js for awhile, I think we've settled on Bacon; now it is just a matter of learning everything I can ;-) Fortunately, most of the straightforward (i.e., clearly reactive) stuff has been very smooth, and now I'm getting into some code that had been previously written using promises (for examples, interfaces to REST API's) and trying to decide how best to convert them (should I leave them as returning promises and wrap them on the outside, or should I convert everything to return observables instead).

**Comment by raimohanska** [comment]
Fixed in the `0.8` branch.

**Comment by raimohanska** [comment]
Decided against this in 2.0
